api/lock.go

```diff
@@ -123,23 +123,12 @@ func (l *Lock) Lock(stopCh <-chan struct{}) (<-chan struct{}, error) {
 	}
 
 	// Check if we need to create a session first
-	l.lockSession = l.opts.Session
```

Comment:
Removing this means that the lock can't be parameterized with a session any longer